tests/providers/apache/beam/hooks/test_beam.py

```diff
@@ -387,32 +388,47 @@ def test_start_go_pipeline_with_binary(self, mock_runner):
 class TestBeamRunner:
     @mock.patch("subprocess.Popen")
     @mock.patch("select.select")
-    def test_beam_wait_for_done_logging(self, mock_select, mock_popen):
+    def test_beam_wait_for_done_logging(self, mock_select, mock_popen, caplog):
+        fake_logger = logging.getLogger("fake-logger")
+
         cmd = ["test", "cmd"]
-        mock_logging = MagicMock()
-        mock_logging.info = MagicMock()
-        mock_logging.warning = MagicMock()
-        mock_proc = MagicMock()
-        mock_proc.stderr = MagicMock()
-        mock_proc.stderr.readlines = MagicMock(return_value=["test\n", "error\n"])
-        mock_stderr_fd = MagicMock()
-        mock_proc.stderr.fileno = MagicMock(return_value=mock_stderr_fd)
-        mock_proc_poll = MagicMock()
-        mock_select.return_value = [[mock_stderr_fd]]
-
-        def poll_resp_error():
-            mock_proc.return_code = 1
-            return True
-
-        mock_proc_poll.side_effect = [None, poll_resp_error]
-        mock_proc.poll = mock_proc_poll
+        mock_proc = MagicMock(name="FakeProc")
+        fake_stderr_fd = MagicMock(name="FakeStderr")
+        fake_stdout_fd = MagicMock(name="FakeStdout")
+
+        mock_proc.stderr = fake_stderr_fd
+        mock_proc.stdout = fake_stdout_fd
+        fake_stderr_fd.readline.side_effect = [
+            b"test-stderr",
+            StopIteration,
+            b"error-stderr",
+            StopIteration,
+            b"other-stderr",
+        ]
+        fake_stdout_fd.readline.side_effect = [b"test-stdout", StopIteration]
+        mock_select.side_effect = [
+            ([fake_stderr_fd], None, None),
+            (None, None, None),
+            ([fake_stderr_fd], None, None),
+        ]
+        mock_proc.poll.side_effect = [None, True]
+        mock_proc.returncode = 1
         mock_popen.return_value = mock_proc
-        with pytest.raises(Exception):
```

Comment:
`B017`: https://docs.astral.sh/ruff/rules/assert-raises-exception/

cc: @hussein-awala I see you add different bugbear rules, so it might be worthwhile also add this one

Reply:
I've already started working on it, I need to finish the open PRs, then I'll open a PR for it. Some of our exceptions are generic in the code, so we'll need to update those as well, but since we're replacing the exception with one of its subclasses, everything will be fine.